app/models/agents/twitter_user_agent.rb

```diff
@@ -95,18 +101,29 @@ def check
       opts = {:count => 200, :include_rts => include_retweets?, :exclude_replies => exclude_replies?, :include_entities => true, :contributor_details => true}
       opts.merge! :since_id => since_id unless since_id.nil?
 
-      # http://rdoc.info/gems/twitter/Twitter/REST/Timelines#user_timeline-instance_method
-      tweets = twitter.user_timeline(interpolated['username'], opts)
+      if choose_home_time_line?
+
+        tweets = twitter.home_timeline(opts)
 
-      tweets.each do |tweet|
-        if tweet.created_at >= starting_at
-          memory['since_id'] = tweet.id if !memory['since_id'] || (tweet.id > memory['since_id'])
+        tweets.each do |tweet|
+          if tweet.created_at >= starting_at
+            memory['since_id'] = tweet.id if !memory['since_id'] || (tweet.id > memory['since_id'])
 
-          create_event :payload => tweet.attrs
+            create_event :payload => tweet.attrs
+          end
         end
-      end
 
-      save!
+      else
+        tweets = twitter.user_timeline(interpolated['username'], opts)
+
+        tweets.each do |tweet|
```

Comment:
Is this loop the same as the one above?  If it is, maybe something like this would work?

```ruby
if choose_home_time_line?
  tweets = twitter.home_timeline(opts)
else
  tweets = twitter.user_timeline(interpolated['username'], opts)
end

tweets.each do |tweet|
  ...
```